training-net/src/Controllers/MoviesController.cs

```diff
@@ -142,13 +142,30 @@ public IActionResult Edit(MovieViewModel mvm)
 
         [HttpGet("Details")]
         public IActionResult Details(int id)
-        {
-            var movie = UnitOfWork.MovieRepository.Get(id);
+        {   
+            var movie = UnitOfWork.MovieRepository.GetMovieWithComments(id);
             if (movie == null)
             {
                 return NotFound();
             }
-            return View(new MovieViewModel { ID = movie.ID, Title = movie.Title, ReleaseDate = movie.ReleaseDate, Genre = movie.Genre, Price = movie.Price });
+            //var comments = UnitOfWork.CommentRepository.GetAll().Where(comment => comment.MovieID == movie.ID).Select(comment =>  new CommentViewModel { ID = movie.ID, Text = comment.Text, Date = comment.Date, Rating = comment.Rating, MovieID = movie.ID });
+             var comments = new List<CommentViewModel>();
+             foreach(var comment in movie.Comments)
+             {
+                 var com = new CommentViewModel { ID = comment.ID, Date = comment.Date, Text = comment.Text, Rating = comment.Rating };
+                 comments.Add(com);
+             }
+            return View(new MovieViewModel { ID = movie.ID, Title = movie.Title, ReleaseDate = movie.ReleaseDate, Genre = movie.Genre, Price = movie.Price, Comments = comments });
+        }
+
+        [HttpPost]
+        public IActionResult AddComment(MovieViewModel mvm)
```

Comment:
although this is working fine.. it's not what you have to do. we need to add a comment to a movie via AJAX, and here is added by a form. we need that the app don't refresh all the page when a comment is added, only add a comment vía AJAX to the DB and show it in the view. research about this in Google and if you cant solve it let me know